ext/semian/sliding_window.c

```diff
@@ -409,24 +418,30 @@ semian_simple_sliding_window_reject(VALUE self)
 
   sem_meta_lock(res->sem_id);
   {
-    // Store these values because we're going to be modifying the buffer.
-    int start = res->shmem->start;
-    int length = res->shmem->length;
-    dprintf("reject! - start:%d end:%d length:%d max_size:%d", res->shmem->start, res->shmem->end, res->shmem->length, res->shmem->max_size);
-
-    int cleared = 0;
-    for (int i = 0; i < length; ++i) {
-      int index = (start + i) % length;
-      int value = res->shmem->data[index];
-      VALUE y = rb_yield(RB_INT2NUM(value));
-      if (RTEST(y)) {
-        if (cleared++ != i) {
-          sem_meta_unlock(res->sem_id);
-          rb_raise(rb_eArgError, "reject! must delete monotonically");
+    semian_simple_sliding_window_shared_t *window = res->shmem;
+    const int start = window->start;
+    const int length = window->length;
+    const int max_size = window->max_size;
+
+    if (max_size && length) {
+      int wptr = (start + length + max_size - 1) % max_size;
```

Comment:
It takes a little bit of reading to grok exactly the approach you're taking here (move elements to the end of the buffer). Maybe a short comment saying what's happening below at a high level could be useful? Up to you.

Reply:
Done.

Reply:
Done.